src/Tasks/PopulateSupportedArchitectures.cs

```diff
@@ -0,0 +1,200 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.Versioning;
+using System.Text;
+using System.Xml;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Shared;
+using Microsoft.Build.Tasks.Deployment.ManifestUtilities;
+using Microsoft.Build.Utilities;
+
+#nullable enable
+
+namespace Microsoft.Build.Tasks
+{
+    /// <summary>
+    /// Generates an application manifest or adds an entry to the existing one when PreferNativeArm64 property is true.
+    /// </summary>
+    [SupportedOSPlatform("windows")]
+    public sealed class PopulateSupportedArchitectures : TaskExtension
+    {
+        private const string supportedArchitectures = "supportedArchitectures";
+        private const string windowsSettings = "windowsSettings";
+        private const string SupportedArchitecturesValue = "amd64 arm64";
+        private const string asmv3Prefix = "asmv3";
+        private const string DefaultManifestName = "default.win32manifest";
+        private const string WindowsSettingsNamespace = "http://schemas.microsoft.com/SMI/2024/WindowsSettings";
+
+        private string _outputPath = string.Empty;
+        private string _generatedManifestFullPath = string.Empty;
+
+        /// <summary>
+        /// Path to the existing application manifest.
+        /// </summary>
+        public string? ApplicationManifestPath { get; set; }
+
+        /// <summary>
+        /// Intermediate output path.
+        /// </summary>
+        [Required]
+        public string OutputPath
```

Comment:
nit: Would you consider renaming to `OutputDirectory` to emphasize that, unlike `ManifestPath`, it is not supposed to contain file name?